packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -0,0 +1,181 @@
+import { MediationRecipientService } from '../routing'
+
+import { OutOfBandMessage } from './OutOfBandMessage'
+
+@scoped(Lifecycle.ContainerScoped)
+export class OutOfBandModule {
+  private connectionService: ConnectionService
+  private mediationRecipientService: MediationRecipientService
+  private disoverFeaturesService: DiscoverFeaturesService
+  private messageSender: MessageSender
+  private eventEmitter: EventEmitter
+
+  public constructor(
+    connectionService: ConnectionService,
+    mediationRecipientService: MediationRecipientService,
+    disoverFeaturesService: DiscoverFeaturesService,
+    messageSender: MessageSender,
+    eventEmitter: EventEmitter
+  ) {
+    this.connectionService = connectionService
+    this.mediationRecipientService = mediationRecipientService
+    this.disoverFeaturesService = disoverFeaturesService
+    this.messageSender = messageSender
+    this.eventEmitter = eventEmitter
+  }
+
+  /**
+   * Creates new connection record and use its keys for out-of-band message that works as a connection invitation.
+   * It uses discover features to find out what handshake protocols the agent supports.
+   *
+   * @param options Optinal attributes contained in out-of-band message
+   * @returns Created connection record and out-of-band message
+   */
+  public async createInvitation(
+    options: OutOfBandMessageOptions
+  ): Promise<{ outOfBandMessage: OutOfBandMessage; connectionRecord: ConnectionRecord }> {
+    // Discover what handshake protocols are supported
+    const queryMessage = new DiscoverFeaturesQueryMessage({
+      query: `*`,
+    })
+    const featuresMessage = await this.disoverFeaturesService.createDisclose(queryMessage)
+    const { protocols } = featuresMessage
+
+    const handshakeProtocols = ['https://didcomm.org/didexchange', 'https://didcomm.org/connections']
+
+    const supportedHandshakeProtocols = protocols
+      .map((p) => p.protocolId.slice(0, -1))
+      .filter((pId) => handshakeProtocols.find((hp) => pId.startsWith(hp)))
+
+    if (supportedHandshakeProtocols.length === 0) {
+      throw new AriesFrameworkError('There is no handshake protocol supported. Agent can not create a connection.')
+    }
+
+    // Create connection.
+    // Eventually, we can create just an OutOfBand record here.
+    // The OOB record can be also used for connection-less communication in general.
+    // Either way, we need to get routing
+    const mediationRecord = await this.mediationRecipientService.discoverMediation()
+    const routing = await this.mediationRecipientService.getRouting(mediationRecord)
+    const { connectionRecord: connectionRecord } = await this.connectionService.createInvitation({
+      routing,
+    })
+
+    const outOfBandMessage = new OutOfBandMessage(options)
+    outOfBandMessage.accept.push('didcomm/aip2;env=rfc587')
+    outOfBandMessage.accept.push('didcomm/aip2;env=rfc19')
```

Comment:
See https://github.com/hyperledger/aries-rfcs/blob/main/features/0044-didcomm-file-and-mime-types/README.md#defined-profiles

We don't support `didcomm/aip2;env=rfc587`, and I think also not `didcomm/aip2;env=rfc19` but not sure